### PR TITLE
QA-120 Refactor upgrade tests

### PR DESCRIPTION
**Pull request by madlabman**

### CODE REVIEW

**Review by dgusakov — DISMISSED** [review]

File: tests/functional/test_adcm_upgrade.py [hunk 1]
```diff
@@ -10,176 +10,77 @@
...
-    with adcm_client(
-        adcm_repo=old_repo,
-        adcm_credentials=adcm_credentials,
-        adcm_tag=old_tag,
-        volumes={volume.name: {'bind': '/adcm/data', 'mode': 'rw'}},
-        init=True,
-    ) as old_adcm_client:
-        bundle = old_adcm_client.upload_from_fs(get_data_dir(__file__, 'cluster_bundle'))
+@pytest.mark.parametrize("image", old_adcm_images(), ids=repr)
+def test_upgrade_adcm(adcm_fs: ADCM, sdk_client_fs: ADCMClient):
```
Comment (vgorkavenko) [inline tests/functional/test_adcm_upgrade.py:39]: Hereinafter. Add `docstring` for test

File: tests/functional/test_adcm_upgrade.py [hunk 2]
```diff
@@ -10,176 +10,77 @@
...
+    hostprovider_bundle = sdk_client_fs.upload_from_fs(get_data_dir(__file__, "hostprovider"))
+    hostprovider = hostprovider_bundle.provider_create(name=f"test_{random_string()}")
+    host = hostprovider.host_create(fqdn=f"test_host_{random_string()}")
+    sdk_client_fs.upload_from_fs(get_data_dir(__file__, "cluster_with_cluster_pass_verify"))
+    cluster_name = f"test_{random_string()}"
+    cluster = sdk_client_fs.cluster_prototype().cluster_create(name=cluster_name)
+    cluster.host_add(host)
+    cluster.config_set_diff(dict(password="q1w2e3r4"))
+    adcm_fs.upgrade(("hub.arenadata.io/adcm/adcm", "latest"))
+    with allure.step("Check that cluster is present"):
```
Comment (vgorkavenko) [inline tests/functional/test_adcm_upgrade.py:62]: Expand step title. Now it does not reflect essence of checks

File: tests/functional/test_adcm_upgrade.py [hunk 3]
```diff
@@ -10,176 +10,77 @@
...
+    host = hostprovider.host_create(fqdn=f"test_host_{random_string()}")
+    sdk_client_fs.upload_from_fs(get_data_dir(__file__, "cluster_with_service_pass_verify"))
+    cluster_name = f"test_{random_string()}"
+    cluster = sdk_client_fs.cluster_prototype().cluster_create(name=cluster_name)
+    cluster.host_add(host)
+    service_name = "PassChecker"
+    service = cluster.service_add(name=service_name)
+    service.config_set_diff(dict(password="q1w2e3r4"))
+    adcm_fs.upgrade(("hub.arenadata.io/adcm/adcm", "latest"))
+    with allure.step('Check cluster'):
```
Comment (vgorkavenko) [inline tests/functional/test_adcm_upgrade.py:81]: Expand step title. Now it does not reflect essence of checks

File: tests/functional/test_adcm_upgrade.py [hunk 4]
```diff
@@ -10,176 +10,77 @@
...
+        assert len(sdk_client_fs.cluster_list()) == 1, "There is no clusters. Expecting one"
```
Comment (dgusakov) [inline tests/functional/test_adcm_upgrade.py:82]:
```suggestion
        assert len(sdk_client_fs.cluster_list()) == 1, "There are no clusters. Expecting one"
```

File: tests/functional/test_adcm_upgrade.py [hunk 5]
```diff
@@ -10,176 +10,77 @@
...
+        cluster = sdk_client_fs.cluster(name=cluster_name)
+        assert len(cluster.service_list()) == 1, "There is no services. Expecting one"
```
Comment (dgusakov) [inline tests/functional/test_adcm_upgrade.py:84]:
```suggestion
        assert len(cluster.service_list()) == 1, "There are no services. Expecting one"
```

File: tests/functional/test_adcm_upgrade.py [hunk 6]
```diff
@@ -10,176 +10,77 @@
...
+    hostprovider = hostprovider_bundle.provider_create(name=f"test_{random_string()}")
+    host = hostprovider.host_create(fqdn=f"test_host_{random_string()}")
+    sdk_client_fs.upload_from_fs(get_data_dir(__file__, "cluster_with_cluster_pass_verify"))
+    cluster_name = f"test_{random_string()}"
+    cluster = sdk_client_fs.cluster_prototype().cluster_create(name=cluster_name)
+    cluster.host_add(host)
+    cluster.config_set_diff(dict(password="q1w2e3r4"))
+    adcm_fs.upgrade(("hub.arenadata.io/adcm/adcm", "latest"))
+    with allure.step("Check that cluster is present"):
+        assert len(sdk_client_fs.cluster_list()) == 1, "There is no clusters. Expecting one"
```
Comment (dgusakov) [inline tests/functional/test_adcm_upgrade.py:63]:
```suggestion
        assert len(sdk_client_fs.cluster_list()) == 1, "There are no clusters. Expecting one"
```

File: tests/functional/test_adcm_upgrade.py [hunk 7]
```diff
@@ -10,176 +10,77 @@
...
-                len(latest_adcm_client.cluster_list()) == 1
-            ), "There is no clusters. Expecting one"
-            cluster = latest_adcm_client.cluster_list()[0]
-            assert len(cluster.service_list()) == 1, "There is no services. Expecting one"
-            service = cluster.service_list()[0]
-            assert service.action(name="check-password").run().wait() == "success"
+        sdk_client_fs.cluster_prototype().cluster_create(name=cluster_name)
+    adcm_fs.upgrade(("hub.arenadata.io/adcm/adcm", "latest"))
+    with allure.step("Check that cluster is present"):
+        assert len(sdk_client_fs.cluster_list()) == 1, "There is no clusters. Expecting one"
```
Comment (dgusakov) [inline tests/functional/test_adcm_upgrade.py:46]:
```suggestion
        assert len(sdk_client_fs.cluster_list()) == 1, "There are no clusters. Expecting one"
```

File: tests/functional/test_adcm_upgrade.py (hunk 2 above)
Comment (madlabman) [reply tests/functional/test_adcm_upgrade.py]: Updated.

File: tests/functional/test_adcm_upgrade.py (hunk 3 above)
Comment (madlabman) [reply tests/functional/test_adcm_upgrade.py]: Done.

**Comment by dgusakov** [comment]
bender build

**Comment by BenderArenadata** [comment]
Hey, @dgusakov! Build has been launched!

**Review by dgusakov — APPROVED** [review]

**Review by vgorkavenko — APPROVED** [review]

**Pull request merged**